### PR TITLE
Add support for virtual environments

### PR DESCRIPTION
**Pull request by iGr33k**
### Description/Motivation/Screenshots ###

When running GDB from virtual environment GDB always loads system's python so GEF can't load modules from the venv required to perform some GEF commands. This patch aims to provide a support for virtual environments so GEF is able to load modules from them. While it is mostly a GDB related problem I thought it would be nice to have this kind of feature in GEF since no other software I use really need such support for venvs.

### How Has This Been Tested? ###

| Architecture |          Yes/No          | Comments                                  |
| ------------ | :----------------------: | ----------------------------------------- |
| x86-32       | :heavy_multiplication_x: |                                    |
| x86-64       | :heavy_check_mark: |                                           |
| ARM          | :heavy_multiplication_x: |                                           |
| AARCH64      | :heavy_multiplication_x: |                                           |
| MIPS         | :heavy_multiplication_x: |                                           |
| POWERPC      | :heavy_multiplication_x: |                                           |
| SPARC        | :heavy_multiplication_x: |                                           |
| RISC-V       | :heavy_multiplication_x: |                                           |
| `make tests` | :heavy_check_mark: |                                           |

### Checklist ###

- [x] My PR was done against the `dev` branch, not `master`.
- [x] My code follows the code style of this project.
- [] My change includes a change to the documentation, if required.
- [] My change adds tests as appropriate.
- [x] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: gef.py [hunk 1]
```diff
@@ -10382,6 +10382,14 @@ def __gef_prompt__(current_prompt):
             site.addsitedir(site_packages_dir)
         except FileNotFoundError:
             pass
+        
+        # when Python virtual enviromrnt is used GDB still loads system Python
+        # so GEF doesn't load site-packages dir from environment
+        pythonbin = which("python3")
+        PREFIX = gef_pystring(subprocess.check_output([pythonbin, '-c', 'import os,sys;print((sys.prefix))'])).strip("\\n")
```
Comment (daniellimws) [inline gef.py:10389]: Why `strip("\\n")`?

**Review by daniellimws — APPROVED** [review]
Tested on virtualenv and it works well! @Grazfather and @hugsy anything from you guys?

File: gef.py (hunk 1 above)
Comment (iGr33k) [reply gef.py:10389]: `sys.base_prefix` comes without `\\n`. Without it I can't validate if GEF is run inside Venv .

File: gef.py [hunk 2]
```diff
@@ -10382,6 +10382,14 @@ def __gef_prompt__(current_prompt):
             site.addsitedir(site_packages_dir)
         except FileNotFoundError:
             pass
+        
+        # when Python virtual enviromrnt is used GDB still loads system Python
```
Comment (Grazfather) [inline gef.py:10386]: typo enviromrnt

```suggestion
        # When using a Python virtual environment, GDB still loads the system Python
```

File: gef.py [hunk 3]
```diff
@@ -10382,6 +10382,14 @@ def __gef_prompt__(current_prompt):
...
+        # so GEF doesn't load site-packages dir from environment
```
Comment (Grazfather) [inline gef.py:10387]: Explain what you're doing to fix it.

File: gef.py [hunk 4]
```diff
@@ -10382,6 +10382,14 @@ def __gef_prompt__(current_prompt):
...
+        pythonbin = which("python3")
+        PREFIX = gef_pystring(subprocess.check_output([pythonbin, '-c', 'import os,sys;print((sys.prefix))'])).strip("\\n")
+        if PREFIX != sys.base_prefix:
+            SITE_PACKAGES_DIR = subprocess.check_output([pythonbin, "-c", "import os,sys;print(os.linesep.join(sys.path).strip())"]).decode("utf-8").split()
```
Comment (Grazfather) [inline gef.py:10391]:
```suggestion
            SITE_PACKAGES_DIR = subprocess.check_output(
                [pythonbin, "-c", "import os,sys;print(os.linesep.join(sys.path).strip())"]).decode("utf-8").split()
```
Let's avoid long lines

File: gef.py (hunk 3 above)
Comment (iGr33k) [reply gef.py:10387]: I tried to explain it in added comments. Hope it is better now.

Comment (daniellimws) [reply gef.py:10387]: Some suggestions/typos:

* `when` -> `When`
* `enviromrnt` -> `environment`
* `system Python` -> `system-installed Python`

File: gef.py [hunk 5]
```diff
@@ -10382,6 +10382,18 @@ def __gef_prompt__(current_prompt):
             site.addsitedir(site_packages_dir)
         except FileNotFoundError:
             pass
+        
+        # When using a Python virtual environment, GDB still loads the system Python
+        # so GEF doesn't load site-packages dir from environment
+        # In order to fix it, from the shell with venv activated we run the python binary, 
+        # take and parse it's path, add path to the current python process using sys.path.extend
```
Comment (Grazfather) [inline gef.py:10389]: `its`, not `it's`.

`add THE path...`

File: gef.py [hunk 6]
```diff
@@ -10382,6 +10382,18 @@ def __gef_prompt__(current_prompt):
...
+        
+        pythonbin = which("python3")
+        PREFIX = gef_pystring(subprocess.check_output([pythonbin, '-c', 'import os,sys;print((sys.prefix))'])).strip("\\n")
+        if PREFIX != sys.base_prefix:
+            SITE_PACKAGES_DIR = subprocess.check_output(
```
Comment (Grazfather) [inline gef.py:10394]: nit: This is a list of DIRs, so `...DIRS`

**Review by Grazfather — COMMENTED** [review]
Looks good, just naming/spelling/grammar

**Comment by Grazfather** [comment]
Nice! Thanks for the PR.

**Pull request merged**